### PR TITLE
[Typescript:Axios] Fix withInterfaces and usSingleRequestObjectParam

### PR DESCRIPTION
**Pull request by dangoslen**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
---

This PR addresses issue #6783 by essentially re-using the same mustache configurations used in the `DefaultApi` as in the `DefaultApiInterface`

Mentioning the TS committee members as per the checklist:

@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov @davidgamero @mkusaka

### CODE REVIEW

File: samples/client/petstore/typescript-axios/builds/with-interfaces-and-with-single-request-param/api.ts [hunk 1]
```diff
@@ -0,0 +1,3318 @@
...
+     *
+     * @summary Updates a pet in the store with form data
+     * @param {number} petId ID of pet that needs to be updated
+     * @param {string} [name] Updated name of the pet
+     * @param {string} [status] Updated status of the pet
+     * @param {*} [options] Override http request option.
+     * @throws {RequiredError}
+     */
+    updatePetWithForm: async (
+      petId: number,
```
Comment (macjohnny) [inline samples/client/petstore/typescript-axios/builds/with-interfaces-and-with-single-request-param/api.ts:629]: shouldnt these be combined into a single request parameters? are the samples up-to-date?

Comment (dangoslen) [reply samples/client/petstore/typescript-axios/builds/with-interfaces-and-with-single-request-param/api.ts:629]: They likely should be. My change was focused on the `*Interface` component that starts on line [1180](https://github.com/OpenAPITools/openapi-generator/pull/19467/files#diff-8bf02f07382b9f8b689c969baf8181bc2d9116c07d108b6405b832185427bb91R1180), but you are right the other components are not updated.

I'll look into updating those as well!

Comment (dangoslen) [reply samples/client/petstore/typescript-axios/builds/with-interfaces-and-with-single-request-param/api.ts:629]: @macjohnny I looked into this a bit and realized to update the ParamCreator and Fp style would be a pretty major change. The main benefit of this change is it aligns the interface to the default API client, which I believe is what most consumers of a generate client would use anyway. 

Just bringing this up as I think there might be some tradeoffs to consider between solving the immediate bug that opened the initial issue (https://github.com/OpenAPITools/openapi-generator/issues/6783) or trying to reset all of the other components in the template

**Review by macjohnny — APPROVED** [review]

**Comment by macjohnny** [comment]
@dangoslen can you re-generate the samples? then we can merge this one

File: samples/client/petstore/typescript-axios/builds/with-interfaces-and-with-single-request-param/api.ts (hunk 1 above)
Comment (macjohnny) [reply samples/client/petstore/typescript-axios/builds/with-interfaces-and-with-single-request-param/api.ts:629]: ok makes sense, lets fix this step by step

**Pull request merged**